### ISSUE TITLE
Roslyn.sln builds successfully, but IntelliSense reports 313884 errors

### ISSUE THREAD

**Issue by lorcanmooney** [dotnet/roslyn#5876]
If I open Roslyn.sln and do a full rebuild, the build succeeds, but over the course of the next 10 minutes or thereabouts, the Error List window slowly fills up with over 300000 intelligence errors. Opening random files shows that syntax highlighting for types is fairly hit-or-miss.  
![intellisense_errors](https://cloud.githubusercontent.com/assets/13181165/10419859/72ce58d8-707c-11e5-9795-55a5e50aa972.png)  
This isn't new, I'm running the Update 1 CTP (14.0.24606.01) now, but I saw similar behaviour in the RTM release.  I haven't seen anything like this in any other solution I've tried, but they're all orders of magnitude smaller than Roslyn.  
I know this isn't much to go on - I'm willing to collect logs, or anything that isn't _too_ destructive.  
Also, anything that involves resizing the Error List window is ... sluggish, but I guess that's to be expected :)

**Comment by Pilchie** [comment]
Can you try:

1. Go to Tools\Options\NuGet Package Manager\General and disable the "Allow NuGet to download missing packages" option.
2. Close VS, and delete the .vs directory beside Roslyn.sln
3. From the root of your Roslyn source tree, run .\nuget restore Roslyn.sln
4. Open VS and see if you get the errors then?

Tagging @jasonmalinowski, @emgarten, and @yishaigalatzer.  We seem to have some issues with the IDE restore, even with 3.2 installed.

**Comment by jasonmalinowski** [comment]
@lorcanmooney: I have an [extension available](https://github.com/jasonmalinowski/roslyn-info-logger/releases) that you can run to create a (really huge) log file that might help us figure out what happened here. If you get a log file, feel free to e-mail it to @Pilchie or myself.

**Comment by lorcanmooney** [comment]
@Pilchie: Unfortunately this had the same outcome. I tried a bunch of other things like a fresh clone, running the msbuild command from the wiki, delete the nugget package cache from my user profile. Sooner or later, they all ended up with the same result.

@jasonmalinowski: I've created the log file based on a fresh clone, and manually restoring the nugget packages into an empty cache. I'll send it along on shortly.

In the meantime, I'll create a clean VM and see what happens there. Maybe comparing the log files will give a clue. Presumably since there aren't more people complaining, this is something specific to my machine/setup. I had various pre-RTM releases installed in the past, so maybe its some leftover cruft from those?

Thanks for the help with this.

**Comment by paul1956** [comment]
I just downloaded the lastest zip and after following @Pilchie instructions I get over 300,000 intelligence errors. Do you want another log? I have never installed a pre-RTM build on my Windows 10 machine.

**Comment by lorcanmooney** [comment]
I did a fresh install in a VM and the issue repros immediately. Here are the steps I went through:
* Installed Windows 8.1 Pro, and then ran multiple iterations of Windows Update to make it current
* Custom install of VS2015 RTM with Git tools, VS SDK and the Windows 10 SDK (plus whatever dependencies they pulled in)
* Configured Windows account as standard user (it also repros under the Administrator account, so I don't think this is a factor)
* Launched VS, chose C# Settings when prompted by the first-run dialog, then disabled the built in nuget restore
* Opened the "Developer command prompt for VS2015"
* Cloned the Roslyn repo onto my desktop
* Ran nuget restore roslyn.sln
* Opened Roslyn.sln. The IntelliSense errors started pouring in almost immediately.
* Right-clicked solution and clicked rebuild. The flood of errors pauses, then starts to grow again once the build is complete.

**Comment by Pilchie** [comment]
@lorcanmooney provided a log last night.  A relevant snippet is below.  This indicated that the IntelliSense build failed, and so the project system didn't tell the IDE about any of the references, files, or options for this project.  Naturally, this is going to cause a lot of downstream errors, since just about everything in Roslyn depends on this project.

I'll set up a clean VM to investigate why this is happening, but a quick guess is that it's failing to run the syntax generator because that project hasn't built yet, and therefore failing the designtime build.

FWIW, my normal development flow is that whenever I make a significant change to my git clone (updating after a couple of days, etc), I
1. Close VS
2. git clean -xdf
3. .\nuget restore Roslyn.sln
4. msbuild /nologo /m /v:m /fl /flp:Verbosity=Normal Roslyn.sln
5. Open Roslyn.sln in VS.

Note that I added the command line build step before opening Roslyn.sln which I admit shouldn't be necessary, but does seem to allow the IntelliSenseBuild to succeed.

```xml
  <project id="(ProjectId, #d0ff1edd-885c-4b4a-b208-55b6e804b22c - %USERPROFILE%\Source\dotnet\roslyn\src\Compilers\CSharp\Portable\CSharpCodeAnalysis.csproj)" name="CSharpCodeAnalysis" assemblyName="CSharpCodeAnalysis" language="C#" path="%USERPROFILE%\Source\dotnet\roslyn\src\Compilers\CSharp\Portable\CSharpCodeAnalysis.csproj">
    <msbuildReferences>
      <ProjectReference Include="..\..\..\Tools\Source\CompilerGeneratorTools\Source\CSharpSyntaxGenerator\CSharpSyntaxGenerator.csproj" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
        <Project>{288089C5-8721-458E-BE3E-78990DAB5E2D}</Project>
        <Name>CSharpSyntaxGenerator</Name>
        <ReferenceOutputAssembly>false</ReferenceOutputAssembly>
      </ProjectReference>
      <ProjectReference Include="..\..\..\Tools\Source\CompilerGeneratorTools\Source\BoundTreeGenerator\CompilersBoundTreeGenerator.csproj" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
        <Project>{02459936-CD2C-4F61-B671-5C518F2A3DDC}</Project>
        <Name>CompilersBoundTreeGenerator</Name>
        <ReferenceOutputAssembly>false</ReferenceOutputAssembly>
      </ProjectReference>
      <ProjectReference Include="..\..\..\Tools\Source\CompilerGeneratorTools\Source\CSharpErrorFactsGenerator\CSharpErrorFactsGenerator.csproj" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
        <Project>{288089C5-8721-458E-BE3E-78990DAB5E2E}</Project>
        <Name>CSharpErrorFactsGenerator</Name>
        <ReferenceOutputAssembly>false</ReferenceOutputAssembly>
      </ProjectReference>
      <ProjectReference Include="..\..\Core\Portable\CodeAnalysis.csproj" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
        <Project>{1EE8CAD3-55F9-4D91-96B2-084641DA9A6C}</Project>
        <Name>CodeAnalysis</Name>
      </ProjectReference>
    </msbuildReferences>
    <workspaceReferences />
    <compilationReferences />
    <compilation objectId="1" assemblyIdentity="CSharpCodeAnalysis, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null">
      <types />
    </compilation>
    <diagnostics />
  </project>
```

**Comment by paul1956** [comment]
I tried the steps above, the msbuild /nologo /m /v:m /fl /flp:Verbosity=Normal Roslyn.sln has multiple warnings and many errors. After restarting Visual Studio I get 32 warnings like below followed by 300,000 errors.
```
The command "C:\Users\Paul\.nuget\packages\FakeSign\0.9.2\tools\FakeSign.exe "C:\Users\Paul\Documents\GitHub\repos\roslyn\Binaries\Obj\Microsoft.DiaSymReader.PortablePdb\Debug\Microsoft.DiaSymReader.PortablePdb.dll"" exited with code 1.	Microsoft.DiaSymReader.PortablePdb		
```

**Comment by Pilchie** [comment]
@paul1956 Can you share the msbuild.log that was created by the msbuild command above?

**Comment by paul1956** [comment]
From the command line build the MSBuild.log file. 
```
https://onedrive.live.com/redir?resid=5BF2F69E9B9FA722!13635&authkey=!AHVmCTXyJvFAc3Y&ithint=file%2clog
```

**Comment by lorcanmooney** [comment]
> ...failing to run the syntax generator because that project hasn't built yet...

@Pilchie That seems like a reasonable explanation, but shouldn't it resolve itself after a successful build in the IDE?

I followed your suggested steps. At least on my VM, that VS2015 dev console is incorrectly configured for the Win10 SDK. The version number in the lib path is missing a trailing `.0` which causes a bunch of warnings. Despite that, the build seems successful, and opening the solution hasn't (yet) produced any IntelliSense errors, and all the syntax highlighting seems to be working correctly.

I'm going to revert back to my physical machine, see what happens, and report back.

**Comment by lorcanmooney** [comment]
That seems to have to have resolved the issue for me. May I suggest updating [Getting the code](https://github.com/dotnet/roslyn/wiki/Building%20Testing%20and%20Debugging#getting-the-code) with that info?

**Comment by paul1956** [comment]
At a minimum I seem to be Missing NuGet Packages, I get imports that are reported as having no members, but that usually means that the package is missing.below are just a few.
```
Imports Roslyn.Test.Utilities
Imports Microsoft.CodeAnalysis.Editor.UnitTests.AutomaticCompletion
Imports Microsoft.CodeAnalysis.Editor.UnitTests.Extensions
Imports Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces

```

**Comment by diryboy** [comment]
I built d0c0f5 using cibuild.cmd and VS today and have some 44 Error and 170 Warnings from Intellisense complaining about the VbiCore project. But switching the filter from `Build + Intellisense` to `Build Only` only give 0 errors and  35 warnings.

The cibuild.cmd went good with all tests passed, while the VS build gave the following output
```
Restoring NuGet packages...
To prevent NuGet from restoring packages during build, open the Visual Studio Options dialog, click on the Package Manager node and uncheck 'Allow NuGet to download missing packages during build.'
Unknown Compatibility Profile: net46.app
Unknown Compatibility Profile: dnxcore50.app
Error occurred while restoring NuGet packages: Unable to find a version of package 'Microsoft.NETCore.Portable.Compatibility' that is compatible with version constraint '[1.0.0, )'.
1>------ Build started: Project: CsiCore, Configuration: Debug Any CPU ------
1>  CsiCore -> C:\Users\paul\Desktop\roslyn\Binaries\Debug\core-clr\csi.exe
========== Build: 1 succeeded, 0 failed, 117 up-to-date, 0 skipped ==========
```

Sample of the message from Intellisense, others look alike:
```
Error	BC30652	Reference required to assembly '<Missing Core Assembly>, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' containing the type 'Object'. Add one to your project.	VbiCore	C:\...\roslyn\src\Interactive\VbiCore\Vbi.vb	Line9
Error	BC30002	Type 'System.String' is not defined.	VbiCore	C:\...\roslyn\src\Interactive\VbiCore\Vbi.vb	Line10
Warning	AnalyzerDependencyConflict	Analyzer assemblies 'C:\...\.nuget\packages\Microsoft.Net.RoslynDiagnostics\1.1.1-beta1-20150818-01\build\..\tools\Roslyn.Diagnostics.Analyzers.dll' and 'C:\...\.nuget\packages\Microsoft.Net.RoslynDiagnostics\1.1.1-beta1-20150814-01\build\..\tools\Roslyn.Diagnostics.Analyzers.dll' both have identity 'Roslyn.Diagnostics.Analyzers, Version=1.1.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' but different contents. Only one will be loaded and analyzers using these assemblies may not run correctly.	Pdb2Xml		Line1
```

Looks we have similar problem, but luckily mine does not block me at the moment,  joining here to get some update in case this block me later.

**Comment by Pilchie** [comment]
`Error occurred while restoring NuGet packages: Unable to find a version of package 'Microsoft.NETCore.Portable.Compatibility' that is compatible with version constraint '[1.0.0, )'.`

That sounds like you have VS NuGet restore enabled at Tools\Options\NuGet Package Manager\General - "Allow NuGet to download missing packages".

For the vbicore issues, they are known.  I have PR #5893 out to fix them, but unfortunately it currently breaks our Mono build.  Hopefully @agocke is going to take a look at that soonish.

**Comment by diryboy** [comment]
@Pilchie Yes, I have VS restore enabled, I can live with this bug anyway. And thanks for the info on VbiCore.

**Comment by Pilchie** [comment]
@lorcanmooney: Unfortunately, a "real" build doesn't trigger a new "IntelliSense build", so it's quite possible that this doesn't rectify itself in the IDE.  I'll take a look to see if I can make our generation targets more resilient for designtime builds.

**Comment by davkean** [comment]
@Pilchie @jasonmalinowski mentioned that restoring Roslyn.sln isn't enough on a clean box. I'm firing up a VM to figure out what's broken here, and what you also have to do.

**Comment by Pilchie** [comment]
@davkean I've already done a bunch of this work, so you can hold off for a while.

**Comment by Pilchie** [comment]
I did some work this morning to get the IntelliSense build to be clean, but even once I do that there are some issues that remain:

1. If Xaml generation fails during the initial design time build, the .g.cs files never get added to the project once they exist.  The Xaml guys are working on that.
2. For some reason, we seem be dropping P2P refs in the workspace here, but I can't figure out why :frowning:.  I will continue to investigate.

**Comment by SamB** [comment]
Hmm, is there some reason that you don't add a `.nuget/NuGet.targets` file to the repository? According to [the Package Restore docs](https://docs.nuget.org/consume/package-restore#user-content-automatic-package-restore-in-visual-studio), this prevents "Automatic Package Restore in Visual Studio" from happening, so users wouldn't need to manually turn that off.

**Comment by yishaigalatzer** [comment]
That's irrelevant. This is for the msbuild based restore. Roslyn doesn't use it
________________________________

**Comment by Pilchie** [comment]
PR for the IntelliSense build fixes mentioned above: #6298

**Comment by coyot** [comment]
I have a problem while building Roslyn, even as you have stated. At the point of restoring nuget I am getting such kind of errors:

C:\repos\roslyn\build\Targets\VSL.Imports.targets(54,3): error MSB4019: The imported project "C:\Program Files (x86)\MSBuild\Microsoft\Portable\v5.0\Microsoft.Portable.CSharp.targets" was not found. Confirm that the path in the <Import>declaration is correct, and that the file exists on disk. [C:\repos\roslyn\src\Interactive\CsiCore\CsiCore.csproj]

Is there a version 5.0 I am not aware of?

**Comment by Pilchie** [comment]
@coyot - the .NETPortable v5.0 targets are only installed by the Win10 tools right now. Can you confirm that you have those installed?

(Note: VS2015 Update 1 will include these targets by default).

**Comment by davkean** [comment]
See the following: https://github.com/dotnet/roslyn/wiki/Getting-Started-on-Visual-Studio-2015

**Comment by coyot** [comment]
@Pilchie, @davkean - I was able to build it, as I wrote in #6316 with some workarounds. But at least I am able to compile features such as #347 ;-)

**Comment by Pilchie** [comment]
Moving to 1.2 to track the rest of this work.

**Comment by eli78435** [comment]
I tried to run roslyn on my machine with the instruction above:
When I run .\nuget restore Roslyn.sln but got a lot of errors like
error MSB4019: The imported project "C:\Program Files (x86)\MSBuild\Microsoft\Portable\v
5.0\  (I didn't have Portable\v5.0 on my machine)

I've updated vs and installed win phone sdk
and now it's works. thanks.

**Comment by davkean** [comment]
@eli78435  See: https://github.com/dotnet/roslyn/wiki/Building-Testing-and-Debugging. In particular, the required VS components.

**Comment by DustinCampbell** [comment]
Moving to analyzers since that's where the work is happening

**Comment by heejaechang** [comment]
dup to https://github.com/dotnet/roslyn/issues/6809

when intellisense build failed, we should let users know what failed and disable full solution analysis.
